File: fronted/src/api/agent.ts
import { get } from './http'

export interface AgentInfo {
  id: number
  name: string
  description?: string
  tags?: string
  icon?: string
  gradient?: string
  sort_order: number
  is_active: boolean
  is_placeholder: boolean
}

export function getAgents(): Promise<AgentInfo[]> {
  return get<AgentInfo[]>('/api/v1/agents/')
}

File: backend/app/models/agent.py
from sqlalchemy import Column, Integer, String, Text, Boolean
from app.db.base_class import Base


class Agent(Base):
    __tablename__ = "agents"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(255), nullable=False)
    description = Column(Text)
    tags = Column(Text)          # JSON array string: '["tag1","tag2"]'
    icon = Column(String(50))
    gradient = Column(String(255))
    sort_order = Column(Integer, default=0)
    is_active = Column(Boolean, default=True)
    is_placeholder = Column(Boolean, default=False)
